docs/rules/prefer-number-is-integer.md

```diff
@@ -0,0 +1,36 @@
+# Prefer `Number.isInteger()` for integer checking
+
+Enforces the use of [Number.isInteger()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) for checking if a number is an integer.
+
+These different implementations have slightly different behaviours.
+
+For example:
+
+```js
+let number = [['1']];
```

Comment:
It's not immediately clear to the user why this is used as an example.

Reply:
Updated :+1: